src/background/services/zkIdentity/index.ts

```diff
@@ -91,101 +195,73 @@ export default class IdentityService {
 
     return true;
   };
 
-  deleteAllIdentities = async (): Promise<boolean> => {
+  private setDefaultIdentity = async (): Promise<void> => {
     const identities = await this.getIdentitiesFromStore();
 
     if (!identities.size) {
-      return false;
+      await this.clearActiveIdentity();
+      return;
     }
 
-    await Promise.all([this.clearActiveIdentity(), this.identitiesStore.clear(), pushMessage(setIdentities([]))]);
-    await this.historyService.trackOperation(OperationType.DELETE_ALL_IDENTITIES, {});
-
-    return true;
+    const identity = identities.keys().next();
+    await this.updateActiveIdentity({ identities, identityCommitment: identity.value as string });
   };
 
-  getActiveIdentity = async (): Promise<ZkIdentityDecorater | undefined> => {
-    const activeIdentityCommitmentCipher = await this.activeIdentityStore.get<string>();
-
-    if (!activeIdentityCommitmentCipher) {
-      return undefined;
-    }
-
-    const activeIdentityCommitment = this.lockService.decrypt(activeIdentityCommitmentCipher);
-    const identities = await this.getIdentitiesFromStore();
-    const identity = identities.get(activeIdentityCommitment);
-
-    if (!identity) {
-      return undefined;
+  private clearActiveIdentity = async (): Promise<void> => {
+    if (!this.activeIdentity) {
+      return;
     }
 
-    this.activeIdentity = ZkIdentityDecorater.genFromSerialized(identity);
+    this.activeIdentity = undefined;
+    await this.writeActiveIdentity("", "");
+  };
 
-    return this.activeIdentity;
+  createIdentityRequest = async (): Promise<void> => {
+    await this.browsercontroller.openPopup({ params: { redirect: Paths.CREATE_IDENTITY } });
   };
 
-  getActiveIdentityData = async (): Promise<SelectedIdentity> => {
-    const identity = await this.getActiveIdentity();
+  createIdentity = async ({
+    strategy,
+    messageSignature,
+    options,
+  }: NewIdentityRequest): Promise<{ status: boolean; identityCommitment?: bigint }> => {
+    if (!strategy) {
+      throw new Error("strategy not provided");
+    }
 
-    return {
-      commitment: identity ? bigintToHex(identity.genIdentityCommitment()) : "",
-      web2Provider: identity?.metadata.web2Provider || "",
+    const numOfIdentites = await this.getNumOfIdentites();
+
+    const config = {
+      ...options,
+      account: options.account ?? "",
+      identityStrategy: strategy,
+      name: options?.name || `Account # ${numOfIdentites}`,
+      messageSignature: strategy === "interrep" ? messageSignature : undefined,
     };
-  };
 
-  getIdentityCommitments = async (): Promise<{ commitments: string[]; identities: Map<string, string> }> => {
-    const identities = await this.getIdentitiesFromStore();
-    const commitments = [...identities.keys()];
+    const identity = this.createNewIdentity(strategy, config);
 
-    return { commitments, identities };
-  };
+    if (!identity) {
+      throw new Error("Identity not created, make sure to check strategy");
```

Comment:
Can you write test to check this branch?

Reply:
I think this condition will never happen.

Reply:
So, let's remove it.